2a. React/2. Mocking/src/components/solvro-projects-combobox-api.test.tsx

```diff
@@ -0,0 +1,178 @@
+        if (loadingText) {
+            expect(loadingText).toBeInTheDocument();
+        }
+
+        await waitFor(() => {
+            expect(screen.getByText('Eventownik')).toBeInTheDocument();
+        });
+    });
+
+    test('calls API with search parameter when searching', async () => {
+        let lastRequestUrl = '';
+        server.use(
+            http.get('https://kurs-z-testowania.deno.dev/projects', ({ request }) => {
+                lastRequestUrl = request.url;
+                const url = new URL(request.url);
+                const search = url.searchParams.get('search');
+
+                let filteredProjects = mockProjects;
+
+                if (search) {
+                    filteredProjects = mockProjects.filter(project =>
+                        project.label.toLowerCase().includes(search.toLowerCase())
+                    );
+                }
+
+                const response: ProjectsResponse = {
+                    projects: filteredProjects,
+                    total: filteredProjects.length,
+                    filters: {
+                        search: search,
+                    },
+                };
+
+                return HttpResponse.json(response);
+            })
+        );
+
+        renderWithQueryClient(<SolvroProjectsComboboxApi />);
+
+        const comboboxTrigger = screen.getByRole('combobox');
+        await userEvent.click(comboboxTrigger);
+        await waitFor(() => {
+            expect(screen.getByText('Eventownik')).toBeInTheDocument();
+        });
+
+        const searchInput = screen.getByPlaceholderText('Wyszukaj projekt...');
+        await userEvent.clear(searchInput);
+        await userEvent.type(searchInput, 'Eventownik');
+
+        await waitFor(() => {
+            expect(lastRequestUrl).toContain('search=Eventownik');
+        });
+    });
+
+    test('selects a project', async () => {
+        renderWithQueryClient(<SolvroProjectsComboboxApi />);
+
+        const comboboxTrigger = screen.getByRole('combobox');
+        await userEvent.click(comboboxTrigger);
+        await waitFor(() => {
+            expect(screen.getByText('Eventownik')).toBeInTheDocument();
+        });
+
+        const projectOption = screen.getByText('Eventownik');
+        await userEvent.click(projectOption);
+
+        await waitFor(() => {
+            expect(comboboxTrigger).toHaveTextContent('Eventownik');
+        });
+    });
+
+    test('handles API error', async () => {
+        server.use(
+            http.get('https://kurs-z-testowania.deno.dev/projects', () => {
+                return new HttpResponse(null, { status: 500 });
+            })
+        );
+
+        renderWithQueryClient(<SolvroProjectsComboboxApi />);
+
+        const comboboxTrigger = screen.getByRole('combobox');
+        await userEvent.click(comboboxTrigger);
+
+        await waitFor(() => {
+            expect(screen.getByText('Błąd podczas ładowania projektów')).toBeInTheDocument();
```

Comment:
I'm not a fan of exact text matching, but it's just a nitpick from my side, and personal preference